crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -296,6 +297,57 @@ where
             status_code: if receipt.success { Some(U64::from(1)) } else { Some(U64::from(0)) },
         };
 
+        // TODO maybe add a helper function for this
+        let prev_tx_hash = match self
+            .transaction_by_block_and_tx_index(
+                meta.block_number,
+                ((meta.index - 1) as usize).into(),
+            )
+            .await?
+        {
+            Some(tx) => tx.hash,
+            None => return Err(EthApiError::Unsupported("Invalid Block or Transaction index")),
+        };
+
+        let prev_receipt = match self.client().receipt_by_hash(prev_tx_hash)? {
+            Some(recpt) => recpt,
+            None => return Err(EthApiError::Unsupported("Previous receipt not found")),
+        };
```

Comment:
we need to fetch all receipts of the block.

I suggest we fetch via `ReceiptProvider::receipts_by_block` this way we can easily outsource this step to the caching layer later ref #1742